### PR TITLE
Issue/36

### PR DESCRIPTION
**Pull request by aloistr**

### CODE REVIEW

File: tests/swe_lun_occult.phpt [hunk 1]
```diff
@@ -16,6 +16,9 @@ $tjd_ut = $rv['tret'][0];
 var_dump($tjd_ut, print_date($tjd_ut));
 echo "swe_lun_occult_where for Venus\n";
 var_dump(swe_lun_occult_where($tjd_ut, SE_VENUS, "", SEFLG_SWIEPH));
+echo "swe_lun_occult_when_loc for Venus\n";
+$rv = swe_lun_occult_when_loc(2454466.5, SE_VENUS, "", SEFLG_SWIEPH, 0, 12.1, 49.0, 330);
```
Comment (kevindecapite) [inline tests/swe_lun_occult.phpt]: The order of the PHP parameters is expected to be:
```
dlsldddl
```

https://github.com/cyjoelchen/php-sweph/pull/42/files#diff-c1e75cac140e6d9d193cd8e4b70223e134f5bb2c0e1604776419d852538db788R2740

...but the documentation _and_ method call in this test seem to place the last `long` before the last 3 `double`s. At least, I am expecting the `0` parameter to represent a non-backwards (i.e. false) search.

Comment (aloistr) [reply tests/swe_lun_occult.phpt]: Fixed

File: swephp.c [hunk 2]
```diff
@@ -2517,6 +2515,8 @@ Get the name of a house system
 =head3 return array
 
       retflag => (int)            ERR or eclipse type
+		  There are the following eclipse types for lunar eclipses:
+		  SE_ECL_TOTAL, SE_ECL_PARTIAL, SE_ECL_PENUMBRAL
```
Comment (kevindecapite) [inline swephp.c:2494]: The docs list the following eclipse types for `swe_lun_occult_where()` function:
```
SE_ECL_TOTAL
SE_ECL_ANNULAR
SE_ECL_TOTAL | SE_ECL_CENTRAL
SE_ECL_TOTAL | SE_ECL_NONCENTRAL
SE_ECL_ANNULAR | SE_ECL_CENTRAL
SE_ECL_ANNULAR | SE_ECL_NONCENTRAL
SE_ECL_PARTIAL
```
I assume there is a reason I'm unaware of which replaces the missing ones with `SE_ECL_PENUMBRAL` instead?

It's also just a doc block so not a significant concern, simply something I noticed.

**Comment by aloistr** [comment]
ready for merge

**Comment by aloistr** [comment]
I used compiler option -Wall which should be turned on by default, to discover many of these issues.

**Comment by kevindecapite** [comment]
Yes, I ran into this when originally tackling the problem with `swe_rise_trans()` not parsing the value of `ipl` correctly. I needed to use `size_t` instead of `int`. I failed to notice that in this debug session.

> But size_t is 64 bit, 8 byte, and if a string length parameter is declared only as int, a memory overwrite can happen

It is good to now know why this issue occurs.

**Comment by aloistr** [comment]
> Yes, I ran into this when originally tackling the problem with `swe_rise_trans()` not parsing the value of `ipl` correctly. I needed to use `size_t` instead of `int`. I failed to notice that in this debug session.
> 
> > But size_t is 64 bit, 8 byte, and if a string length parameter is declared only as int, a memory overwrite can happen
> 
> It is good to now know why this issue occurs.

Actually, long is 64 bit, and so it makes a difference whether long or int are used. long is essential.
size_t is also 64 bit.

It is important that the programmer checks the parameter list carefully. I was not aware of the issue when I added new functions, and made errors. Numerous errors were also in the older code.
I have now checked the parameter list for all calls of zend_parse_parameters().

**Pull request merged**